### PR TITLE
Added possibiliy of sorting tasks and actions in workflows

### PR DESCRIPTION
**Pull request by konadrian**
Added possibiliy of sorting tasks and actions in workflows
SortWorkflows
SortTasks

### CODE REVIEW

**Comment by ghost** [comment]
## Black Duck Security Report

Merging #16028 into developer will not change security risk.

[Click here to see full report](https://copilot.blackducksoftware.com/github/repos/YetiForceCompany/YetiForceCRM/pull-requests/16028)

File: layouts/basic/modules/Settings/Workflows/ListViewHeader.tpl [hunk 1]
```diff
@@ -6,67 +6,68 @@
...
-			<div class="col-lg-3 btn-toolbar ml-0 mt-1 mt-lg-0">
-				<select class="select2 form-control" id="moduleFilter">
-					<option value="">{\App\Language::translate('LBL_ALL', $QUALIFIED_MODULE)}</option>
-					{foreach item=MODULE_MODEL key=TAB_ID from=$SUPPORTED_MODULE_MODELS}
-						<option {if !empty($SOURCE_MODULE) && $SOURCE_MODULE eq $MODULE_MODEL->getName()} selected="" {/if}
+						<i class="fas fa-download"></i>
+					</button>
+				</div>
+				<div class="col-lg-3 btn-toolbar ml-0 mt-1 mt-lg-0">
+					<select class="select2 form-control js-module-filter" id="moduleFilter">
```
Comment (rskrzypczak) [inline layouts/basic/modules/Settings/Workflows/ListViewHeader.tpl]: js-module-filter ?

File: modules/Settings/Vtiger/views/List.php [hunk 2]
```diff
@@ -38,6 +38,10 @@ public function preProcess(App\Request $request, $display = true)
 	public function process(App\Request $request)
 	{
 		$viewer = $this->getViewer($request);
+		if (!$request->isEmpty('sourceModule')) {
+			$sourceModule = $request->getByType('sourceModule', 2);
```
Comment (rskrzypczak) [inline modules/Settings/Vtiger/views/List.php]: 2 => \App\P..::..

File: modules/com_vtiger_workflow/VTWorkflowManager.php [hunk 3]
```diff
@@ -132,7 +132,8 @@ public function save(Workflow $workflow)
 				'schdayofweek' => $wf->schdayofweek,
 				'schannualdates' => $wf->schannualdates,
 				'nexttrigger_time' => empty($wf->nexttrigger_time) ? null : $wf->nexttrigger_time,
-				'params' => empty($wf->params) ? null : $wf->params
+				'params' => empty($wf->params) ? null : $wf->params,
+				'sequence' => $wf->sequence,
```
Comment (rskrzypczak) [inline modules/com_vtiger_workflow/VTWorkflowManager.php]:
```suggestion
				'sequence' => (int) $wf->sequence,
```

File: public_html/layouts/basic/modules/Settings/Workflows/resources/List.js [hunk 4]
```diff
@@ -72,23 +72,25 @@ Settings_Vtiger_List_Js(
...
 				jQuery('#pageToJump').val('1');
 				jQuery('#orderBy').val('');
 				jQuery('#sortOrder').val('');
-				var params = {
+				let params = {
 					module: app.getModuleName(),
 					parent: app.getParentModuleName(),
-					sourceModule: jQuery(e.currentTarget).val()
+					sourceModule: jQuery(e.currentTarget).val(),
+					orderby: 'sequence'
```
Comment (rskrzypczak) [inline public_html/layouts/basic/modules/Settings/Workflows/resources/List.js:86]: ?

File: public_html/layouts/basic/modules/Settings/Workflows/resources/List.js [hunk 5]
```diff
@@ -123,10 +128,139 @@ Settings_Vtiger_List_Js(
...
+				.fail(function () {
+					app.showNotify({
+						text: app.vtranslate('JS_UNEXPECTED_ERROR'),
+						type: 'error'
+					});
+				});
+		},
+		/**
+		 * Get workflows from list
+		 * @returns [array]
```
Comment (rskrzypczak) [inline public_html/layouts/basic/modules/Settings/Workflows/resources/List.js]: [] ?

**Review by rskrzypczak — CHANGES_REQUESTED** [review]

**Pull request merged**